lib/matplotlib/figure.py

```diff
@@ -157,6 +158,59 @@ def __contains__(self, a):
         return a in self.as_list()
 
 
+class _AxesStack(object):
+    """Lightweight stack that tracks Axes in a Figure.
+    """
+
+    def __init__(self):
+        # We maintain a list of (creation_index, key, axes) tuples.
+        self._items = []
+        self._created = 0
+
+    def as_list(self):
+        """Copy of the list of axes, in the order of insertion.
+        """
+        return [ax for _, _, ax in sorted(self._items)]
+
+    def get(self, key):
+        """Find the axes corresponding to a key; defaults to `None`.
+        """
+        return next((ax for _, k, ax in self._items if k == key), None)
+
+    def current_key_axes(self):
+        """Return the topmost key, axes pair, or `None, None` if empty.
```

Comment:
Should add some parentheses because right now it can read as either "a key" or "an axes pair" or `None, None`, but it's really "(key, axes) pair" or `None, None`.